lib/abci/handlers/initChainHandlerFactory.js

```diff
@@ -0,0 +1,67 @@
+const {
+  abci: {
+    ResponseInitChain,
+  },
+} = require('abci/types');
+const { ChainLock } = require('@dashevo/dashcore-lib');
+const waitForEvent = require('../../util/waitForEvent');
+/**
+ * Init Chain ABCI handler
+ *
+ * @param {LatestCoreChainLock} latestCoreChainLock
+ * @param {BlockchainState} blockchainState
+ * @param {ZMQClient} coreZMQClient
+ * @param {RpcClient} coreRpcClient
+ * @param {BaseLogger} logger
+ *
+ * @return {initChainHandler}
+ */
+function initChainHandlerFactory(
+  latestCoreChainLock,
+  blockchainState,
+  coreZMQClient,
+  coreRpcClient,
+  logger,
+) {
+  /**
+   * @typedef endBlockHandler
+   *
+   * @param {abci.RequestInitChain} request
+   * @return {Promise<abci.ResponseInitChain>}
+   */
+  async function initChainHandler() {
+    logger.info('Init chain');
+
+    // We need to retrieve latest chainlock from our fully synced DashCore instance
+    const rpcBestChainLockResponse = await coreRpcClient.getBestChainLock();
```

Comment:
As descrived in taks we need to obtain initial chainlock before we start ABCI server so it should be implemented in scripts/abci